### ISSUE TITLE
Lockdown Microsoft.ML.Parquet public surface

### ISSUE THREAD

**Issue by TomFinley** [dotnet/machinelearning#2274]
We ought to internalize as much of that assembly to those things we want to be public, and finalize its public API. In particular, making it resemble the other methods for creation of data views out of files (similar to text loading and binary file loading) would probably be the focus. Having it be what we call an `IDataReader` instead of the old legacy `IDataLoader` would be necessary.

If we do not have time for that, then perhaps making everything in this internal or else making the `Microsoft.ML.Parquet` nuget an experimental version is an answer.

**Comment by eerhardt** [comment]
NOTE: We don't ship the ML.Parquet package right now. So this work is lower priority. And if we don't ship ML.Parquet in `v1.0`, we can postpone this work until after `v1.0`.

**Comment by TomFinley** [comment]
Huh. You're right, I search for [Microsoft.ML.Parquet](https://www.nuget.org/packages?q=Microsoft.ML.Parquet) on nuget.org and I see nothing. That's a bit of a surprise. (Just curious, what controls whether one of our packages actually gets shipped or not?)

The reason why I'm curious @eerhardt is we have this here:

https://github.com/dotnet/machinelearning/blob/0e80d6c07580cea5ce072fd7cba02d7171dc61f4/src/Microsoft.ML.Parquet/Microsoft.ML.Parquet.csproj#L5

But it is clearly not posted. But when I wrote new nuget packages, I don't recall ever doing anything special to make them ship, so I'm wondering if people have just been doing that work for me and not telling me. 😄

**Comment by eerhardt** [comment]
> (Just curious, what controls whether one of our packages actually gets shipped or not?)

Uploading to nuget.org is a manual process, which is done by @shauheen.

**Comment by rogancarr** [comment]
I had this added to ML.NET for an internal project. I'm okay if it doesn't hit the public space till post-1.0. It might be better, as I imagine there might be things we will need to chase around a bit before we want to release.

**Comment by codemzs** [comment]
@eerhardt @TomFinley @rogancarr @shauheen  Microsoft.ML.Parquet doesn't expose anything and just contains internal stuff .... this issue can be closed.
